src/common/workspace-store.ts

```diff
@@ -67,14 +71,20 @@ export class WorkspaceStore extends BaseStore<WorkspaceStoreModel> {
   }
 
   @action
-  saveWorkspace(workspace: Workspace) {
-    const id = workspace.id;
+  saveWorkspace(workspace: Workspace): { workspace?: Workspace, error?: string} {
+    const { id, name } = workspace;
```

Comment:
name should be trimmed for leading/trailing spaces (and probably add a unit test for `" "`)

Reply:
Great catch! Thanks for pointing out. Added more test cases.